java/ql/src/experimental/Security/CWE/CWE-094/SpringViewManipulationLib.qll

```diff
@@ -0,0 +1,30 @@
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+import SpringFrameworkLib
+
+/**
+ * A taint-tracking configuration for unsafe user input
+ * that can lead to Spring View Manipulation vulnerabilities.
+ */
+class SpringViewManipulationConfig extends TaintTracking::Configuration {
+  SpringViewManipulationConfig() { this = "Spring View Manipulation Config" }
+
+  override predicate isSource(DataFlow::Node source) {
+    source instanceof RemoteFlowSource or
+    source instanceof WebRequestSource
+  }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof SpringViewManipulationSink }
+}
+
+/**
+ * A sink for Spring View Manipulation vulnerabilities,
+ */
+class SpringViewManipulationSink extends DataFlow::ExprNode {
+  SpringViewManipulationSink() {
+    exists(ReturnStmt r, SpringRequestMappingMethod m |
+      r.getResult() = this.asExpr() and m.getBody().getAStmt() = r and not m.isResponseBody()
```

Comment:
Please include Portlet and Servlet versions of [`ModelAndView`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/ModelAndView.html) which contains multiple sinks in its constructor and `setViewName`.

Also, include the [implicit view resolution cases](https://github.com/veracode-research/spring-view-manipulation/#even-more-magic)

Reply:
Good point, but I think it makes sense to add that separate query part of this PR though